power_grid_model_c/power_grid_model/include/power_grid_model/calculation_parameters.hpp

```diff
@@ -92,41 +93,41 @@ struct MathModelTopology {
     std::vector<double> phase_shift;
     std::vector<BranchIdx> branch_bus_idx;
     std::vector<BranchIdx> fill_in;
-    IdxVector source_bus_indptr;
-    IdxVector shunt_bus_indptr;
-    IdxVector load_gen_bus_indptr;
+    DenseGroupedIdxVector sources_per_bus;
+    SparseGroupedIdxVector shunts_per_bus;
+    SparseGroupedIdxVector load_gens_per_bus;
     std::vector<LoadGenType> load_gen_type;
-    IdxVector voltage_sensor_indptr;
-    IdxVector source_power_sensor_indptr;      // indptr of the source
-    IdxVector load_gen_power_sensor_indptr;    // indptr of the load_gen
-    IdxVector shunt_power_sensor_indptr;       // indptr of the shunt
-    IdxVector branch_from_power_sensor_indptr; // indptr of the branch
-    IdxVector branch_to_power_sensor_indptr;   // indptr of the branch
-    IdxVector bus_power_sensor_indptr;         // indptr of the bus
+    SparseGroupedIdxVector voltage_sensors_per_bus;
+    SparseGroupedIdxVector power_sensors_per_source;
+    SparseGroupedIdxVector power_sensors_per_load_gen;
+    SparseGroupedIdxVector power_sensors_per_shunt;
+    SparseGroupedIdxVector power_sensors_per_branch_from;
+    SparseGroupedIdxVector power_sensors_per_branch_to;
+    SparseGroupedIdxVector power_sensors_per_bus;
```

Comment:
check which ones need to be `DenseGroupedIdxVector` and which ones need to be `SparseGroupedIdxVector`

Reply:
In the grids that I see, shunts are never present.
All the sensors are never multiple at each `measured_object`. Hence it can be DenseGroupedIdxVector.

I guess this needs a brainstorm though. There might be application cases in future where this is not so.

Reply:
cfr. discussion during standup: out of scope of this PR